actors/builtin/miner/miner_state.go

```diff
@@ -1018,6 +1018,19 @@ func (st *State) AdvanceDeadline(store adt.Store, currEpoch abi.ChainEpoch) (*Ad
 	// we still use dlInfo.Last().
 	dlInfo := st.DeadlineInfo(currEpoch)
 
+	// Return early if the proving period hasn't started. While actors v2
+	// sets the proving period start into the past so this case can never
+	// happen, v1:
+	//
+	// 1. Sets the proving period in the future.
+	// 2. Schedules the first cron event one epoch _before_ the proving
+	//    period start.
+	//
+	// At this point, no proofs have been submitted so we can't check them.
+	if !dlInfo.PeriodStarted() {
+		return new(AdvanceDeadlineResult), nil
```

Comment:
So, technically there could be existing faulty power here if a miner reports it. But lotus will never actually report faulty power before the start of the proving period, and I'm not sure if it makes sense to fine miners for that power either.